src/dsql/parse.y

```diff
@@ -2903,6 +2925,13 @@ alter_function_clause
 			$$->alter = true;
 			$$->create = false;
 		}
+	| change_opt_function_clause
```

Comment:
Instead of separate branch you could make whole AS part optional. As a bonus security attributes would become uneatable as well.

Reply:
> Instead of separate branch you could make whole AS part optional. As a bonus security attributes would become uneatable as well.

You mean make psql-body and external-body optional, right? Then you will have to write:
alter function funcname returns type deterministic / not deterministic; 
Accordingly, the "return <type>" part should also be made optional?

Reply:
Why not? It would be standard deviation but quite acceptable, IMHO. But let's wait for others' opinion. It was just an idea, not request from me.

Reply:
`function_clause` is shared between `CREATE` and `ALTER`, so without a separate branch it would be possible to write `CREATE FUNCTION <funcname> DETERMINISTIC`. I'm not sure this is what we want, even if handled internally with some dedicated error.